packages/coding-agent/src/tui/tui-renderer.ts

```diff
@@ -1952,3 +2109,32 @@ export class TuiRenderer {
 		}
 	}
 }
+
+/**
+ * Kill a process and all its children (cross-platform)
+ */
+function killProcessTree(pid: number): void {
```

Comment:
This should likely go into shell-config.ts, which should likely be renamed to shell.ts, and its functions should likely be reused by the bash tool.